specification/cloudhealth/CloudHealth.Management/main.tsp

```diff
@@ -0,0 +1,843 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using OpenAPI;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+
+@armProviderNamespace(ProviderName)
+@service(#{ title: ProviderName })
+@armCommonTypesVersion(CommonTypes.Versions.v6)
+@versioned(Versions)
+namespace Microsoft.CloudHealth;
+
+@doc("API Versions")
+enum Versions {
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v6)
+  @doc("2025-05-01-preview")
+  v2025_05_01_preview: "2025-05-01-preview",
+}
+
+alias ProviderName = "Microsoft.CloudHealth";
+
+// Regex for our health model resource name validations. Between 3 and 44 characters, alphanumeric and hyphens, must start with an alphabetic character and end with an alphanumeric character
+alias modelResourceNameRegex = "^[a-zA-Z][a-zA-Z0-9-]{1,42}[a-zA-Z0-9]$";
+
+@pattern(modelResourceNameRegex)
+@doc("String with regex validation to match the health model resource name requirements")
+scalar stringModelResourceName extends string;
+
+// Regex for all our proxy resource name validations. Up to 260 characters, alphanumeric and hyphens, must start and end with an alphanumeric character
+alias proxyResourceNameRegex = "^[a-zA-Z0-9][a-zA-Z0-9-]{1,258}[a-zA-Z0-9]$";
+
+@pattern(proxyResourceNameRegex)
+@doc("String with regex validation to match the proxy resource name requirements")
+scalar stringResourceName extends string;
```

Comment:
>stringResourceName [](http://example.com/codeflow?start=7&length=18)

considered calling this a proxyResourceName instead?

Reply:
or having one for each type of resource?

Reply:
see other comment.

Reply:
changed